src/main/scala/nl.knaw.dans.easy.deposit/docs/DDM.scala

```diff
@@ -161,22 +162,15 @@ object DDM extends SchemedXml with DebugEnhancedLogging {
   private implicit class RichElem(val elem: Elem) extends AnyVal {
     /** @param str the desired label, optionally with name space prefix */
     @throws[InvalidDocumentException]("when str is not a valid XML label (has more than one ':')")
-    def withLabel(str: String): Elem = {
-      str.split(":") match {
-        case Array(label) => elem.copy(label = label)
-        case Array(prefix, label) => elem.copy(prefix = prefix, label = label)
-        case a => throw invalidDatasetMetadataException(new IllegalArgumentException(
-          s"expecting (label) or (prefix:label); got [${ a.mkString(":") }] to adjust the <${ elem.label }> of ${ trim(elem) }"
-        ))
-      }
-    }
+    def withLabel(str: String): Elem = withLabel(Some(str))
 
-    def withLabel[T](someEnum: Option[T]): Elem = {
-      someEnum.map(_.toString.split(":")) match {
+    @throws[InvalidDocumentException]("when maybeVal does not contain a valid XML label (its .toString has more than one ':')")
+    def withLabel[T](maybeVal: Option[T]): Elem = {
```

Comment:
it would be more logical if you'd implement `def withLabel[T](maybeVal: Option[T])` in terms of `def withLabel(str: String)` rather than the other way around. Now you're packing and unpacking too many times.